### ISSUE TITLE
Make Engine without lifetime.

### ISSUE THREAD

**Issue by azoyan** [rsonquery/rsonpath#613]
I need to store compiled once engines in struct
```rust

#[derive(Deserialize, Serialize, Debug, Clone)]
pub struct RateLimitBodyJsonPath {
    pub engines: Vec<RsonpathEngine>,
}
```
but engine should keep reference with lifetime to query.

Is it possbile to make special Engine that pass query by value once and compile it to engine, and then use just engine to handle incoming json?

**Comment by V0ldek** [comment]
A version of this is actually on my branch for #117 UTF-8 support which is still WIP. I'll try to upstream just the lifetime change sooner without the full #117 feature.

**Comment by V0ldek** [comment]
This is released in v0.9.2, `RsonpathEngine` is now lifetimeless.

**Comment by azoyan** [comment]
Thank you!
